drivers/ethernet/eth_stm32_hal.c

```diff
@@ -846,6 +849,12 @@ static int eth_stm32_hal_set_config(const struct device *dev,
 				     sizeof(dev_data->mac_addr),
 				     NET_LINK_ETHERNET);
 		return 0;
+#if defined(CONFIG_NET_PROMISCUOUS_MODE)
+	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
+		heth->Instance->MACFFR |= config->promisc_mode ? ETH_MACFFR_PM : ~ETH_MACFFR_PM;
+
+		return 0;
+#endif
```

Comment:
```suggestion
	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
#if defined(CONFIG_NET_PROMISCUOUS_MODE)
#if defined(CONFIG_SOC_SERIES_STM32H7X)
		heth->Instance->MACPFR |= config->promisc_mode ? ETH_MACPFR_PR : ~ETH_MACPFR_PR;
#else
		heth->Instance->MACFFR |= config->promisc_mode ? ETH_MACFFR_PM : ~ETH_MACFFR_PM;
#endif  /* CONFIG_SOC_SERIES_STM32H7X */
		ret = 0;
#endif /* CONFIG_NET_PROMISCUOUS_MODE */
		break;
```

Changes here so far:

1.  case should be always active
2. return -ENOSUP if `ETHERNET_CONFIG_TYPE_PROMISC_MODE` is given, but not enabled
3. add support for stm32h7 also

Reply:
Thanks!
This must have been overlooked in review and obviously never disabled by us (or just lucky)
happy with all your suggestions and thanks again